express_relay/programs/express_relay/tests/test_permission.rs

```diff
@@ -0,0 +1,103 @@
+pub mod helpers;
+
+use anchor_lang::prelude::*;
+use solana_program_test::{ProgramTest, tokio};
+use solana_sdk::{account::Account, signature::Keypair, signer::Signer};
+use express_relay::state::{FEE_SPLIT_PRECISION, SEED_EXPRESS_RELAY_FEES, SEED_METADATA};
+use helpers::{initialize, permission};
+
+#[tokio::test]
+async fn test_permission() {
+    let mut program_test = ProgramTest::new(
+        "express_relay",
+        express_relay::id(),
+        None,
+    );
+
+    let payer = Keypair::new();
+    let admin = Keypair::new();
+    let relayer_signer = Keypair::new();
+    let fee_receiver_relayer = Keypair::new().pubkey();
+    let split_protocol_default: u64 = 2000;
+    let split_relayer: u64 = 5000;
+
+    let searcher = Keypair::new();
+
+    let protocol = Keypair::new().pubkey();
+    let fee_receiver_protocol = Pubkey::find_program_address(&[SEED_EXPRESS_RELAY_FEES], &protocol).0;
+
+    program_test.add_account(
+        payer.pubkey(),
+        Account {
```

Comment:
we can reduce the duplicate logic here. the Account structure is the same in all these `add_account`s